horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -982,4 +979,131 @@ public QueryResult queryDataSet(Integer datasetId, String jsonpath, boolean arra
       }
       return result;
    }
+
+   private void transform(Run run, String role) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+      Collection<RunServiceImpl.RunSchema> runSchemas = getSchemas(run, role);
+      Test test = Test.findById(run.testid);
+      String data = run.data.toString();
+      JsonNode node = Util.toJsonNode(data);
+      List<JsonNode> nodesToProcess = new ArrayList<JsonNode>();
+
+      if (runSchemas.isEmpty()) {
+         new DataSet(Instant.now(), Instant.now(), "",
+            test.id, Util.toJsonNode(EMPTY_ARRAY), run, 0, run.owner, run.access)
+            .persist();
+      } else {
+         runSchemas.forEach( s -> {
+            if (hasMatch(s, node)) {
+               nodesToProcess.add(node);
+            }
+            Schema schema = Schema.find("uri = ?1", s.uri).singleResult();
+            List<Transformer> transformers = Transformer.find("schema_id = ?1", schema.id).list();
```

Comment:
Applicable transformers should be listed in `Test.transformers`; the transformation should not apply all transformations for given schema.

Reply:
ok

Reply:
but then what's to determine which schema has which transformations to apply ?